transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -338,6 +369,11 @@ public boolean writeInbound(Object... msgs) {
             return isNotEmpty(inboundMessages);
         }
 
+        if (!config().isAutoRead()) {
+            Collections.addAll(tempInboundMessages(), msgs);
```

Comment:
only call this if `tempInboundMessage != null`

Reply:
Since `tempInboundMessages()` will always return a instance, I remove null check here.

Reply:
I think you still should use the null check to reduce allocations if not needed.